program/src/deposit.rs

```diff
@@ -73,44 +82,71 @@ fn access_control(req: AccessControlRequest) -> Result<(), FundError> {
   if !depositor_authority_acc_info.is_signer {
     return Err(FundErrorCode::Unauthorized)?;
   }
+  {
+    // let rent = access_control::rent(rent_acc_info)?;
+    let _ = access_control::fund(fund_acc_info, program_id)?;
+    let _ = access_control::vault_join(
+      vault_acc_info,
+      vault_authority_acc_info,
+      fund_acc_info,
+      program_id,
+    )?;
 
-  // let rent = access_control::rent(rent_acc_info)?;
-  let fund = access_control::fund(fund_acc_info, program_id)?;
-  let _ = access_control::vault_join(
-    vault_acc_info,
-    vault_authority_acc_info,
-    fund_acc_info,
-    program_id,
-  )?;
+    let _ = access_control::check_balance(fund_acc_info, amount)?;
```

Comment:
should check if the fund is still open for depoists

Reply:
can a fund choose to stop accepting deposits?

Reply:
for now on the first withdrawal it closes the deposits. Adding another tx to explicitly close funds.